lib/travis/build/script/go.rb

```diff
@@ -110,10 +110,10 @@ def uses_make
           
           # see https://golang.org/doc/go1.6#go_command
           def uses_15_vendoring
-            if (go_version == 'go1' || comparable_go_version < Gem::Version.new('1.5'))
+            if (go_version == 'go1' || (go_version != 'tip' && comparable_go_version < Gem::Version.new('1.5')))
```

Comment:
Personally, `go_version == 'tip'` and `go_version.is_tip?` are equally clear to me. And `go_version == 'go1'` appears just as many times, but has no helper.

Three things about this:

1. You're going to be maintaining this long after I'm gone, so I'm happy to use `is_tip?` or whatever, if you'd prefer.
2. If you want me to use `is_tip?`, should I also refactor at the same time to use `is_go1?` instead of `go_version == 'go1'`, for consistency?
3. Why are we keeping `'go1'` around, anyways? `'tip'` I understand, as it has no semver version associated with it. But is there any harm in just setting `go1` to be `1.0.3` or whatever in `comparable_go_version` and removing all the logic around it?